lib/src/configuration.dart

```diff
@@ -133,6 +133,12 @@ class Configuration {
   /// by the [path] you  property. This property is ignored if the directory defined by [path] allow FIFO special files.
   String get fifoFilesFallbackPath => realmCore.getConfigFifoPath(this);
   set fifoFilesFallbackPath(String value) => realmCore.setConfigFifoPath(this, value);
+
+  Function(Realm realm)? get initialDataCallback => _initialDataCallback;
```

Comment:
This looks a bit odd to me.

When dealing with callbacks on the Dart side I expect it either maps to a Future that you can await (if it is a one time thing), or Stream you can listen to, if it can occur multiple times.

So instead of a callback you can register on the `Configuration`, I would expect a `Future` on the `Realm`, something like `Future<void> get initialData => _initialDataCompleter.future` where `Completer<void> _initialDataCompleter`.. this `Completer` (think TaskCompletionSource) is then parsed through to native and back again in the `userdata` parameter.

Reply:
This would be complicated slightly if we allowed configs to be shared between realms - as core does, but we currently don't allow that in Dart.

Reply:
Maybe I misunderstand the purpose of this callback. I thought it was a way to ensure that you would not block while the realm is setup to be ready for use?

Reply:
Why would you need to populate the realm before returning the instance. You may as well do that after?

Reply:
It's a niche use case, but it's something that only happens once - when the Realm is created, which is not something you can easily do with the SDK API as there's no difference between opening an existing Realm vs creating a new one. An example use case would be if you wanted to populate some demo data in your app the first time it's opened and then allow the user to delete it if they want to (think a note taking app with a prepopulated note explaining some of the app functionality).

Reply:
Why not just check if db is empty, and then populate it? 

But what you are saying is that:
```
final r = Realm();
await r.initialData; 
```
would actually wait forever on restart of the app? Because the callback really signifies that the db file has been created, not that the realm is wired up.

Well, I guess that makes many of my comments mood.

Reply:
Well.. leave a crumb in the db that the user cannot delete. Or check if the realm db path exists first, to see if it is the first.. Anyway, the API is there, and now we support it in dart as well :-)